File: ScriptGraph/TemplateScriptNode/Arccos/CGArccos.js
/**
 * @file CGArccos.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGArccos.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGArccos extends BaseNode {
  constructor() {
    super();
  }

  getOutput() {
    return Math.acos(this.inputs[0]());
  }
}

exports.CGArccos = CGArccos;

File: ScriptGraph/TemplateScriptNode/Arccos/CGArccos.lua
local CGArccos = CGArccos or {}
CGArccos.__index = CGArccos

function CGArccos.new()
    local self = setmetatable({}, CGArccos)
    self.inputs = {}
    return self
end

function CGArccos:setInput(index, func)
    self.inputs[index] = func
end

function CGArccos:getOutput(index)
    return math.acos(self.inputs[0]())
end

return CGArccos
